### ISSUE TITLE
Cannot use udp for neighbors

### ISSUE THREAD

**Issue by pr4xx** [bluedigits/iota-node#1]
Hey there, thanks for this image!
I got a problem where I cannot sync with a neighbor when we use udp. My partner got multiple neighbors with udp without any trouble.  Only if we switch both to tcp, then our nodes sync.
We both turned off firewall and made sure we can communicate via udp tested with iperf.
Maybe you know some way?

**Comment by bluedigits** [comment]
Hey Nico,

Thank you for your message. I'm glad you find the image useful.

Can you please provide your docker run command? I had some hiccups with udp ports exposed by docker until I found out that I had to specify '--net=host'.

Does netstat show your open port as udp or udp6?

I'm sure we can figure it out.

Best,
Merlin
Am 26.08.2017 um 22:13 schrieb Nico <notifications@github.com<mailto:notifications@github.com>>:


Hey there, thanks for this image!
I got a problem where I cannot sync with a neighbor when we use udp. My partner got multiple neighbors with udp without any trouble. Only if we switch both to tcp, then our nodes sync.
We both turned off firewall and made sure we can communicate via udp tested with iperf.
Maybe you know some way?

**Comment by pr4xx** [comment]
This is the part in my docker-compose file:
```
services:
  iota:
    image: bluedigits/iota-node
    container_name: iota
    ports:
      - "14265:14265"
      - "14777:14777/udp"
      - "15777:15777"
    volumes:
      - ./iota/data:/iri/data
      - ./iota/neighbors:/iri/conf/neighbors
    restart: always
```

This is my docker-compose ps:
```
iota                    /docker-entrypoint.sh            Up      0.0.0.0:14265->14265/tcp, 0.0.0.0:14777->14777/udp, 0.0.0.0:15777->15777/tcp
```

This is my netstat:
```
udp6       0      0 :::14777                :::*
```

Seems like udp6 (ipv6?) is used here. This may be a problem?

**Comment by bluedigits** [comment]
I had the same issue but for me the problem disappeared as soon as I managed docker to not expose the specified port as udp6 anymore but udp (ipv4 instead).

So, short answer: Yeah, that might be the problem.
Am 27.08.2017 um 13:34 schrieb Nico <notifications@github.com<mailto:notifications@github.com>>:


This is the part in my docker-compose file:

services:
  iota:
    image: bluedigits/iota-node
    container_name: iota
    ports:
      - "14265:14265"
      - "14777:14777/udp"
      - "15777:15777"
    volumes:
      - ./iota/data:/iri/data
      - ./iota/neighbors:/iri/conf/neighbors
    restart: always


This is my docker-compose ps:

iota                    /docker-entrypoint.sh            Up      0.0.0.0:14265->14265/tcp, 0.0.0.0:14777->14777/udp, 0.0.0.0:15777->15777/tcp


This is my netstat:

udp6       0      0 :::14777                :::*


Seems like udp6 (ipv6?) is used here. This may be a problem?

**Comment by pr4xx** [comment]
Seems like adding `network_mode: "host"` to my docker-compose.yml fixed the problem. Yeah!
Thanks for your help! Next time I really should do what the docs state ;)

**Comment by bluedigits** [comment]
Try to specify network_mode: "host" below your iota service spec

Am 27.08.2017 um 13:34 schrieb Nico <notifications@github.com<mailto:notifications@github.com>>:


This is the part in my docker-compose file:

services:
  iota:
    image: bluedigits/iota-node
    container_name: iota
    ports:
      - "14265:14265"
      - "14777:14777/udp"
      - "15777:15777"
    volumes:
      - ./iota/data:/iri/data
      - ./iota/neighbors:/iri/conf/neighbors
    restart: always


This is my docker-compose ps:

iota                    /docker-entrypoint.sh            Up      0.0.0.0:14265->14265/tcp, 0.0.0.0:14777->14777/udp, 0.0.0.0:15777->15777/tcp


This is my netstat:

udp6       0      0 :::14777                :::*


Seems like udp6 (ipv6?) is used here. This may be a problem?

**Comment by bluedigits** [comment]
Ah you've been faster ;) Glad it helped!

Am 27.08.2017 um 13:50 schrieb Nico <notifications@github.com<mailto:notifications@github.com>>:


Closed #1<https://github.com/bluedigits/iota-node/issues/1>.
